File: 백준/1644.cpp
#include<iostream>
#include<vector>
using namespace std;
vector<long long> prime;
void get_prime(long long);
//12164kb	28ms
int main()
{
	cin.tie(0);
	ios::sync_with_stdio(0);
	//최대 400만까지 이므로 int -> long long
	long long N;
	cin >> N;
	get_prime(N);
	long long index_from=0, index_to = 0;
	long long res = 0;
	long long tmp_sum = 0;
	//투 포인터
	while(index_from <= index_to)
	{
		if (tmp_sum <= N)
		{
			if (index_to < prime.size())
			{
				tmp_sum += prime[index_to];
				index_to++;
			}
			else
				break;
		}
			
		else
		{
			tmp_sum -= prime[index_from];
			index_from++;
		}
		if (tmp_sum == N)
			res++;
	}
	cout << res;
	return 0;
}
void get_prime(long long n)//에라토스테네스의 체
{
	bool *check = new bool[n + 2]{ 0, };
	
	for (long long i = 2; i <= n; i++)
	{
		if (check[i])
			continue;
		prime.push_back(i);
		for (long long j = i*i; j <= n; j += i)
			check[j] = 1;
	}
	delete check;
}

File: 백준/1644.py
N = int(input())
check = [0 for _ in range(N+2)]
prime = []
prime_ap = prime.append
tmp_sum=0
s,e = 0,0
cnt =0
if N <2:
    print(0)
else:
    #소수 구하기
    #소수 구하면서 중간중간에도 연산하면 222552kb	720ms
    #그냥 소수 다 구하고 연산하면 	222424kb	616ms
    for i in range(2, N+1):
        if check[i]:
            continue
        prime_ap(i)
        # if s>e:
        #     break
        # if tmp_sum > N:
        #     tmp_sum -= prime[s]
        #     s+=1
        # else:
        #     tmp_sum += prime[e]
        #     e+=1
        # if tmp_sum == N:
        #     cnt += 1

        for j in range(i*i,N+1,i):
            check[j] = 1
    while s<=e:
        if tmp_sum > N:
            tmp_sum -= prime[s]
            s+=1
        elif e == len(prime):
            break
        else:
            tmp_sum += prime[e]
            e+=1
        if tmp_sum == N:
            cnt += 1
    print(cnt)